examples/vue-instantsearch/src/App.vue

```diff
@@ -0,0 +1,428 @@
+
+  const querySuggestionsPlugin = createQuerySuggestionsPlugin({
+    searchClient,
+    indexName: 'instant_search_demo_query_suggestions',
+    getSearchParams() {
+      const currentCategory = getInstantSearchCurrentCategory();
+
+      if (!currentCategory) {
+        return recentSearchesPlugin.data.getAlgoliaSearchParams({
+          hitsPerPage: 6,
+        });
+      }
+
+      return recentSearchesPlugin.data.getAlgoliaSearchParams({
+        hitsPerPage: 3,
+        facetFilters: [
+          `${INSTANT_SEARCH_INDEX_NAME}.facets.exact_matches.categories.value:-${currentCategory}`,
+        ],
+      });
+    },
+    categoryAttribute: [
+      'instant_search',
+      'facets',
+      'exact_matches',
+      INSTANT_SEARCH_HIERARCHICAL_ATTRIBUTE,
+    ],
+    transformSource({ source }) {
+      const currentCategory = getInstantSearchCurrentCategory();
+
+      return {
+        ...source,
+        sourceId: 'querySuggestionsPlugin',
+        onSelect({ item }) {
+          setInstantSearchUiState({
+            query: item.query,
+            category: item.__autocomplete_qsCategory,
+          });
+        },
+        getItems(params) {
+          if (!params.state.query) {
+            return [];
+          }
+
+          return source.getItems(params);
+        },
+        templates: {
+          ...source.templates,
+          header({ items }) {
+            if (!currentCategory || items.length === 0) {
+              return null;
+            }
+
+            return (
+              <Fragment>
+                <span className="aa-SourceHeaderTitle">
+                  In other categories
+                </span>
+                <span className="aa-SourceHeaderLine" />
+              </Fragment>
+            );
+          },
+        },
+      };
+    },
+  });
+
+  return {
+    subscribe() {
+      const initialState =
+        instantSearchInstance.mainIndex.getHelper()?.state || {};
+
+      autocompleteInstance = autocomplete({
+        container: '#autocomplete',
+        placeholder: 'Search for products',
+        detachedMediaQuery: 'none',
+        openOnFocus: true,
+        plugins: [
+          recentSearchesPlugin,
+          querySuggestionPluginInCategory,
+          querySuggestionsPlugin,
+        ],
+        initialState: { query: initialState.query || '' },
+        onSubmit({ state }) {
+          setInstantSearchUiState({ query: state.query });
+        },
+        onReset() {
+          setInstantSearchUiState({ query: '', resetCategory: true });
+        },
+        onStateChange({ prevState, state }) {
+          if (prevState.query !== state.query) {
+            setInstantSearchUiState({ query: state.query });
+          }
+        },
+        renderer: {
+          createElement: h,
+          Fragment,
+        },
+        render({ children }, root) {
+          render(children, root);
+        },
+      });
+    },
+    unsubscribe() {
+      autocompleteInstance?.destroy();
+    },
+  };
+}
```

Comment:
The idea I went with for this example is to wrap all of our autocomplete logic in a [middleware](https://www.algolia.com/doc/api-reference/widgets/middleware/vue/).

This allows us to instanciate autocomplete after vue-instantsearch, and to have access to `instantSearchInstance`, which is critical for `uiState` interaction.